pyproject.toml

```
@@ -0,0 +1,131 @@
+[build-system]
+requires = ["setuptools>=42", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "pylate"
+# The version is dynamically loaded from pylate.__version__
+dynamic = ["version"]
+description = "A library for training and retrieval with ColBERT"
+readme = "README.md"
+authors = [
+    {name = "LightON"},
+]
+requires-python = ">=3.8"
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "Operating System :: OS Independent",
+]
+dependencies = [
+    "sentence-transformers == 4.0.2",
+    "datasets >= 2.20.0",
+    "accelerate >= 0.31.0",
+    "voyager >= 2.0.9",
+    "sqlitedict >= 2.1.0",
+    "pandas >= 2.2.1",
+    "transformers == 4.48.2",
+]
+
+[project.urls]
+"Homepage" = "https://github.com/lightonai/pylate"
+
+[project.optional-dependencies]
+eval = [
+    "ranx >= 0.3.16",
+    "beir >= 2.0.0",
+]
+api = [
+    "fastapi >= 0.114.1",
+    "uvicorn >= 0.30.6",
+    "batched >= 0.1.2",
+]
+dev = [
+    "ruff >= 0.4.9",
+    "pytest-cov >= 5.0.0",
+    "pytest-xdist >=3.6.0",
+    "pytest-rerunfailures >= 15.0.0",
+    "pytest >= 8.2.1",
+    "pandas >= 2.2.1",
+    "mkdocs-material == 9.5.32",
+    "mkdocs-awesome-pages-plugin == 2.9.3",
+    "mkdocs-jupyter == 0.24.8",
+    "mkdocs_charts_plugin == 0.0.10",
+    "numpydoc == 1.8.0",
+    "einops>=0.8.1",
+    "pre-commit>=4.1.0",
+]
```

Maybe split test, dev, docs dependencies?
```suggestion
dev = [
    "ruff >= 0.4.9",
    "einops>=0.8.1",
    "pre-commit>=4.1.0",
]
doc = [
    "mkdocs-material == 9.5.32",
    "mkdocs-awesome-pages-plugin == 2.9.3",
    "mkdocs-jupyter == 0.24.8",
    "mkdocs_charts_plugin == 0.0.10",
    "numpydoc == 1.8.0",
]
test = [
    "pytest-cov >= 5.0.0",
    "pytest-xdist >=3.6.0",
    "pytest-rerunfailures >= 15.0.0",
    "pytest >= 8.2.1",
    "pandas >= 2.2.1",
]
```